src/ui/Card.tsx

```diff
@@ -75,104 +66,47 @@ export function Card(props: CardProps): JSX.Element {
-        [
-          [
-            cond(
-              greaterThan(abs(panX), MIN_PAN_FOR_ACTION),
-              // Swiping.
-              [
-                set(config.toValue, multiply(divide(panX, abs(panX)), WINDOW_WIDTH)),
-                [set(currentAction, CurrentAction.Swiping)],
-                cond(greaterThan(panX, 0), call([], swipeRight), call([], swipeLeft)),
-              ],
-              // Releasing without a swipe.
-              [set(config.toValue, 0)],
-            ),
-            cond(
-              not(clockRunning(clock)),
-              // Start spring animation.
-              [
-                set(springState.finished, 0),
-                set(springState.time, 0),
-                set(springState.position, panX),
-                set(springState.velocity, velX),
-                startClock(clock),
-              ],
-            ),
-          ],
-          // TODO: This animation takes ~1.5s to finish after the card is
-          // already offscreen. Figure out how to abort early rather than
-          // waiting for the spring to settle.
-          cond(springState.finished, [
-            // Finished spring animation.
-            cond(eq(currentAction, CurrentAction.Swiping), [
-              call([], () => props.onCardOffScreen({ removeThreadId: props.thread.id() })),
-            ]),
-            set(currentAction, CurrentAction.None),
-            stopClock(clock),
-          ]),
-          Animated.spring(clock, springState, config),
-          springState.position,
-        ],
-        [panX],
-      ),
-    ]);
-  }
-
-  const panX = Animated.useValue(0 + props.xOffset);
-  const velocityX = Animated.useValue(0);
-  const panState = Animated.useValue(State.UNDETERMINED);
-  const [clock] = useState(new Clock());
-  const panDrawX = useAnimation(panX, velocityX, clock, panState);
+  const handleGesture = Animated.event([{ nativeEvent: { translationX: pan.x } }], {
+    useNativeDriver: true,
+  });
 
-  const handleGesture = Animated.event(
-    [
-      {
-        nativeEvent: {
-          translationX: panX,
-          velocityX: velocityX,
-          state: panState,
-        },
-      },
-    ],
-    {
-      useNativeDriver: true,
-    },
-  );
+  const handleGestureStateChange = async (evt: GestureHandlerStateChangeEvent): Promise<void> => {
+    const nativeEvent = (evt.nativeEvent as unknown) as GestureHandlerGestureEventNativeEvent &
+      PanGestureHandlerEventExtra;
+    if (evt.nativeEvent.state === State.END) {
+      if (Math.abs(nativeEvent.translationX) < MIN_PAN_FOR_ACTION) {
+        Animated.spring(pan, {
+          ...SPRING_CONFIG,
+          toValue: { x: 0, y: 0 },
+          velocity: nativeEvent.velocityX,
+          useNativeDriver: true,
+        }).start();
+      } else {
+        if (!messages.length) {
+          // TODO: Make it so that the UI isn't swipeable until we've loaded message data.
+          assertNotReached('Have not loaded message data yet.');
```

Comment:
nit: This could be:
`assert(!messages.length, 'Have not loaded message data yet.');`

Reply:
Will do, but as a followup.